crates/bevy_solari/src/realtime/restir_di.wgsl

```diff
@@ -0,0 +1,117 @@
+#import bevy_core_pipeline::tonemapping::tonemapping_luminance as luminance
```

Comment:
If you want a full approval from me, I'm going to need a bit of a comment-based walkthrough of each of the steps that we're taking inside of this shader. I understand the conceptual approach of restir, but I'm having trouble mapping it to this code because of my shader noobiness.